pkg/logs/internal/util/testutils/hostname_mock.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package testutils
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+)
+
+type mockHostnameService struct {
+	name string
+}
+
+func (m *mockHostnameService) Get(_ context.Context) (string, error) {
```

Comment:
From @gh123man in #21987:
> [We already have a hostname mock ](https://github.com/DataDog/datadog-agent/blob/ff99122acb83f841a058d4ed7ba59f76e73a862e/comp/core/hostname/hostnameimpl/service_mock.go#L36)- can we re-use this instead of duplicating all the code?

Reply:
The existing hostname mock can't be used in `pkg/logs` because it's in `comp/core/hostnameimpl` which can't be converted to a module. We could move this mock into `comp/core/hostnameinterface`, but we would also need to move the `Mock` definition to do this.

Reply:
Please try to avoid duplicate code.

Reply:
I moved `service_mock.go` and `component_mock.go` to `hostnameinterface` so that I could remove the duplicate mock in testutils, let me know if this works